spec/formalism/form/coercion_spec.rb

```diff
@@ -0,0 +1,189 @@
+# frozen_string_literal: true
+
+## This is unit test, created to simplify testing of different coercions
+## You can find integration test in spec/formalism/form/fields_spec.rb
+## (Formalism::Form -> .field -> coersion)
+##
+## In tests `eql` used instead of `eq` to determine result type
+describe Formalism.const_get(:Coercion) do
+	let(:value) {}
+	let(:type) {}
+
+	subject(:coercion_result) { described_class.new(value, type: type).result }
+
+	shared_examples 'it parses nil' do |expected_result = nil|
+		let(:value) { nil }
+		it { is_expected.to eql(expected_result) }
+	end
+
+	shared_examples 'it parses empty string' do |expected_result = nil|
+		let(:value) { '' }
+		it { is_expected.to eql(expected_result) }
+	end
+
+	shared_examples 'it parses empty array' do |expected_result = nil|
+		let(:value) { [] }
+		it { is_expected.to eql(expected_result) }
+	end
+
+	shared_examples 'it parses empty hash' do |expected_result = nil|
+		let(:value) { {} }
+		it { is_expected.to eql(expected_result) }
+	end
+
+	describe 'coercion to String' do
```

Comment:
I think, `coercion` there is duplicated. In `doc` format output will be `Form::Coersion coersion to String`. I think, `to String` is OK, or `#to_string` is even better.

Reply:
`describe 'to String'` don't sound, so i used `'#to_string'`